evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -240,6 +250,95 @@ fn remove_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes,
     Ok(Bytes::new())
 }
 
+/// Converts a serde_json::Value to an abi::Token
+/// The function is designed to run recursively, so that in case of an object
+/// it will call itself to convert each of it's value and encode the whole as a
+/// Tuple
+fn value_to_token(value: &Value) -> Result<Token, Token> {
+    if value.is_boolean() {
+        Ok(Token::Bool(value.as_bool().unwrap()))
+    } else if value.is_string() {
+        let val = value.as_str().unwrap();
+        // If it can decoded as an address, it's an address
+        if let Ok(addr) = H160::from_str(val) {
+            Ok(Token::Address(addr))
+        } else {
+            Ok(Token::String(val.to_owned()))
+        }
+    } else if value.is_u64() {
+        Ok(Token::Uint(value.as_u64().unwrap().into()))
+    } else if value.is_i64() {
+        Ok(Token::Int(value.as_i64().unwrap().into()))
+    } else if value.is_array() {
+        let arr = value.as_array().unwrap();
+        Ok(Token::Array(arr.iter().map(|val| value_to_token(val).unwrap()).collect::<Vec<Token>>()))
+    } else if value.is_object() {
+        let values = value
+            .as_object()
+            .unwrap()
+            .values()
+            .map(|val| value_to_token(val).unwrap())
+            .collect::<Vec<Token>>();
+        Ok(Token::Tuple(values))
+    } else {
+        Err(Token::String("Could not decode field".to_owned()))
+    }
+}
+/// Parses a JSON and returns a single value, an array or an entire JSON object encoded as tuple.
+/// As the JSON object is parsed serially, with the keys ordered alphabetically, they must be
+/// deserialized in the same order. That means that the solidity `struct` should order it's fields
+/// alphabetically and not by efficient packing or some other taxonomy.
+fn parse_json(_state: &mut Cheatcodes, json: &str, key: &str) -> Result<Bytes, Bytes> {
+    let values: Value = jsonpath_rust::JsonPathFinder::from_str(json, key)?.find();
+    // values is an array of items. Depending on the JsonPath key, they
+    // can be many or a single item. An item can be a single value or
+    // an entire JSON object.
+    let res = values
+        .as_array()
+        .ok_or(util::encode_error("JsonPath did not return an array"))?
+        .iter()
+        .map(|inner| {
+            let val = inner;
+            value_to_token(val).unwrap()
+        })
+        .collect::<Vec<Token>>();
+    // encode the bytes as the 'bytes' solidity type
+    let abi_encoded = abi::encode(&[Token::Bytes(abi::encode(&res))]);
+    Ok(abi_encoded.into())
+}
+
+fn write_json(
+    state: &mut Cheatcodes,
+    keys: &Vec<String>,
+    values: &Vec<String>,
+    path: impl AsRef<Path>,
+    overwrite: &bool,
```

Comment:
don't borrow Copy types
```suggestion
    overwrite: bool,
```